packages/frontend/src/providers/App/AppProvider.tsx

```diff
@@ -1,11 +1,23 @@
-import { type FC } from 'react';
+import { useState, type FC } from 'react';
+import { EMBED_KEY } from '../../ee/providers/Embed/types';
 import useHealth from '../../hooks/health/useHealth';
 import useUser from '../../hooks/user/useUser';
 import AppProviderContext from './context';
 
+/**
+ * We start with an empty embed context to avoid any issues for users/admins that use both the Lightdash app UI
+ * and the Lightdash embedded experience.
+ * If a embedded user enters the app, the JWT will be set in {@link EmbedProvider}
+ */
 const AppProvider: FC<React.PropsWithChildren<{}>> = ({ children }) => {
     const health = useHealth();
     const user = useUser(!!health?.data?.isAuthenticated);
+    const [isInitialized, setIsInitialized] = useState(false);
+
+    if (!isInitialized) {
+        localStorage.setItem(EMBED_KEY, JSON.stringify({}));
```

Comment:
why do we need this ? why not using  sessionStorage instead ?

Reply:
Like we have now a `getFromLocalStorage` function, I'd create a `setLocalStorage` function (we might have something already) , and centralize both functions in the same file, just in case we want to chagne types, or replace local with session.

Reply:
That's a great point about putting the setter in the same place! 

We need this because the app will get confused if we are trying to use both cookies and JWTs. Thinking more now though, what we should do is return early in the JWT middleware if we already have a session user.